addons/purchase/models/purchase.py

```diff
@@ -1020,8 +1033,9 @@ def _convert_to_tax_base_line_dict(self):
             currency=self.order_id.currency_id,
             product=self.product_id,
             taxes=self.taxes_id,
-            price_unit=self.price_unit,
+            price_unit=self.original_unit_price,
```

Comment:
Since our discount apply on tax excluded price, so here you don't really need `original_unit_price` and `discount`. you could just pass `price_unit` as it's the final tax excluded price.

Reply:
Hello @yhu-odoo 
Since the "price unit" is already a discounted price and if we pass this "price_unit" then it will recalculate with "discount" for "price_subtotal" which is not correct , 
The same concern also extends to the creation of bills.

Reply:
ok